react/features/local-recording/components/LocalRecordingInfoDialog.js

```diff
@@ -0,0 +1,332 @@
+                            {`${t('localRecording.moderator')}:`}
+                        </span>
+                        <span className = 'spacer'>&nbsp;</span>
+                        <span className = 'info-value'>
+                            { isModerator
+                                ? t('localRecording.yes')
+                                : t('localRecording.no') }
+                        </span>
+                    </div>
+                    { isOn && <div>
+                        <span className = 'info-label'>
+                            {`${t('localRecording.duration')}:`}
+                        </span>
+                        <span className = 'spacer'>&nbsp;</span>
+                        <span className = 'info-value'>
+                            { durationString }
+                        </span>
+                    </div>
+                    }
+                    {isOn
+                    && <div>
+                        <span className = 'info-label'>
+                            {`${t('localRecording.encoding')}:`}
+                        </span>
+                        <span className = 'spacer'>&nbsp;</span>
+                        <span className = 'info-value'>
+                            { encodingFormat }
+                        </span>
+                    </div>
+                    }
+                    {
+                        isModerator
+                        && <div>
+                            <div>
+                                <span className = 'info-label'>
+                                    {`${t('localRecording.participantStats')}:`}
+                                </span>
+                            </div>
+                            { this.renderStats() }
+                        </div>
+                    }
+                    {
+                        isModerator
+                            && <div className = 'info-dialog-action-links'>
+                                <div className = 'info-dialog-action-link'>
+                                    {isOn ? <a
+                                        onClick = { this._onStop }>
+                                        { t('localRecording.stop') }
+                                    </a>
+                                        : <a
+                                            onClick = { this._onStart }>
+                                            { t('localRecording.start') }
+                                        </a>
+
+                                    }
+                                </div>
+                            </div>
+                    }
+                </div>
+            </div>
+        );
+    }
+
+    /**
+     * Creates a duration string "HH:MM:SS" from two Date objects.
+     *
+     * @param {Date} now - Current time.
+     * @param {Date} prev - Previous time, the time to be subtracted.
+     * @returns {string}
+     */
+    _getDuration(now, prev) {
```

Comment:
Is it possible for prev to be undefined? Does that look okay in the UI if it's undefined?

Reply:
Hmm. Yes. And moment.js will output the string `invalid dates` which is displayed on the UI.

Reply:
Added a null check. https://github.com/jitsi/jitsi-meet/pull/3223/commits/5fc7078f5da6494473843c68345e74927bb3bad6#diff-c79e43537b94c65e3956d4a7bd366a52R272